internal/api/impl/auth/oidc.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/securecookie"
+	"github.com/labstack/echo/v4"
+	"github.com/perses/perses/internal/api/shared"
+	"github.com/perses/perses/internal/api/shared/crypto"
+	"github.com/perses/perses/internal/api/shared/route"
+	"github.com/perses/perses/pkg/model/api/config"
+	"github.com/sirupsen/logrus"
+	"github.com/zitadel/oidc/v3/pkg/client/rp"
+	httphelper "github.com/zitadel/oidc/v3/pkg/http"
+	"github.com/zitadel/oidc/v3/pkg/oidc"
+)
+
+type oIDCEndpoint struct {
+	relyingParty    rp.RelyingParty
+	jwt             crypto.JWT
+	tokenManagement tokenManagement
+	slugID          string
+	urlParams       map[string]string
+}
+
+func newOIDCEndpoint(provider config.OIDCProvider, jwt crypto.JWT) (authEndpoint, error) {
+	// As the cookie is used only at login time, we don't need a persistent value here.
+	key := securecookie.GenerateRandomKey(16)
```

Comment:
Could you elaborate on that please? Can you explain why we need to generate a cookie for the authentication ?

Reply:
I don't have much more details, if any cookie is generated by the lib here, we are not using it directly and setting random key doesn't change anything... 

Oh maybe I'm configuring a cookie handler for nothing here, let me check

Reply:
:smile: I removed the line and it still work... nevermind we'll remove it

Reply:
okok I checked twice and no it is used during the authorization code flow, then removed. So this is what I said we don´t need it persistent.
Before redirect to authorize provider's endpoint, the OIDC lib save the state and a pkce code challlenge into cookies, this is then reused to verify the state and code received into our code exchange callback, and removed. 

I will elaborate into the comment itself.